library/platform_util.c

```diff
@@ -65,3 +72,45 @@ void mbedtls_platform_zeroize( void *buf, size_t len )
     memset_func( buf, 0, len );
 }
 #endif /* MBEDTLS_PLATFORM_ZEROIZE_ALT */
+
+#if defined(MBEDTLS_HAVE_TIME_DATE) && !defined(MBEDTLS_PLATFORM_GMTIME_ALT)
+#include <time.h>
+#if !defined(_WIN32) && (defined(__unix__) || \
+    (defined(__APPLE__) && defined(__MACH__)))
+#include <unistd.h>
+#if !defined(_POSIX_VERSION) || _POSIX_C_SOURCE > _POSIX_THREAD_SAFE_FUNCTIONS
+#define PLATFORM_UTIL_USE_GMTIME
+#endif /* !_POSIX_VERSION || _POSIX_C_SOURCE > _POSIX_THREAD_SAFE_FUNCTIONS */
+#endif /* !_WIN32 && (__unix__ || (__APPLE__ && __MACH__)) */
+
+struct tm *mbedtls_platform_gmtime( const mbedtls_time_t *tt,
```

Comment:
Since, we are creating an abstraction we should abstract ABI as well. We shouldn't return `struct tm *` from an `mbedtls_` abstract function. If we intend to keep the ABI same as `gmtime`, still there should be a `typedef`.

Reply:
I see what you mean, but I am not sure this is necessary in this case. The struct tm is from the C standard and Mbed TLS assumes that struct tm is as in the standard because it directly accesses the fields of that struct: https://github.com/ARMmbed/mbedtls/blob/development/library/x509.c#L914. What are your thoughts?